tools/proofers/preview.js

```diff
@@ -107,6 +117,65 @@ $(function () {
             }
         }
 
+        const reNoteStart = /\[\*\*/g;
+
+        function checkProoferNotes() {
+            // look for [** then look for ] skipping matched [ ]
+            let result, closeIndex;
+            while(null !== (result = reNoteStart.exec(txt))) {
+                closeIndex = findClose(txt, reNoteStart.lastIndex);
+                if (0 === closeIndex) {
+                    // no ] found
+                    reportIssue(result.index, 3, "noCloseBrack", 1);
+                    badParse();
+                    return;
+                } else {
+                    // continue search from the ] we found
+                    reNoteStart.lastIndex = closeIndex;
+                }
+            }
+        }
+
+        let notes = [];
+
+        // make text with notes removed, put them in an array with indexes into the new text
+        function removeAllNotes(txt) {
+            let beginIndex = 0;
+            let txtOut = "";
+            let result, noteStartIndex, noteEndIndex;
+            // start of note relative to txtOut
+            let outNoteStart = 0;
+            const maxIndex = txt.length - 1;
+            reNoteStart.lastIndex = 0;
+            // we've already checked that notes are correctly terminated
+            // find start of note
+            while (null != (result = reNoteStart.exec(txt))) {
+                noteStartIndex = result.index;
+                noteEndIndex = findClose(txt, reNoteStart.lastIndex);
+                // copy text to start of note
+                txtOut += txt.slice(beginIndex, noteStartIndex);
+                outNoteStart += noteStartIndex - beginIndex;
+                // if note starts at beginning of a line and ends at the end of a line
+                // remove any following nl character also so doesn't count as a blank line and include the nl in the note
+                if(((noteStartIndex === 0) || (txt.charAt(noteStartIndex - 1) === "\n")) && ((noteEndIndex === maxIndex) || (txt.charAt(noteEndIndex + 1) === "\n"))) {
```

Comment:
maybe add some newlines to group the conditions more clearly:
```
if(((noteStartIndex === 0) || (txt.charAt(noteStartIndex - 1) === "\n")) &&
   ((noteEndIndex === maxIndex) || (txt.charAt(noteEndIndex + 1) === "\n"))) {
```

Reply:
good idea.